### PR TITLE
[Fixes] Swapped the values for tenantId and deploymentSpId

### PR DESCRIPTION
**Pull request by thomas-aure**
# Description

In the enforcedTokensList hashtable the key deploymentSpId currently holds the value of the ARM_TENANT_ID and the key tenantId hold the value of the DEPLOYMENT_SP_ID.   This PR swaps these values to their corresponding value. 

Solves #1457 

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 1]
```diff
@@ -44,22 +44,22 @@
 
 parameters:
   # Pipeline-related parameters
-  serviceConnection: '$(serviceConnection)'
-  poolName: '$(poolName)'
-  vmImage: '$(vmImage)'
+  serviceConnection: "$(serviceConnection)"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:47]: Please search and replace `"` -> `'`. There is a setting to do this in VSCode `"yaml.format.singleQuote": true` which we have in the .vscode/settings.json file.

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]
Other then the quotes, again, a good catch from you Thomas!

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml (hunk 1 above)
Comment (MariusStorhaug) [reply .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:47]:
```suggestion
  serviceConnection: '$(serviceConnection)'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 2]
```diff
@@ -44,22 +44,22 @@
...
+  poolName: "$(poolName)"
+  vmImage: "$(vmImage)"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:49]:
```suggestion
  poolName: '$(poolName)'
  vmImage: '$(vmImage)'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 3]
```diff
@@ -44,22 +44,22 @@
...
   defaultJobTimeoutInMinutes: 120
-  checkoutRepositories: ''
+  checkoutRepositories: ""
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:51]:
```suggestion
  checkoutRepositories: ''
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 4]
```diff
@@ -44,22 +44,22 @@
...
-  subscriptionId: '$(ARM_SUBSCRIPTION_ID)'
-  managementGroupId: '$(ARM_MGMTGROUP_ID)'
-  modulesRepository: '$(modulesRepository)'
+  modulePath: "$(modulePath)"
+  parametersRepository: "$(Build.Repository.Name)"
+  location: "$(location)"
+  resourceGroupName: "$(resourceGroupName)"
+  subscriptionId: "$(ARM_SUBSCRIPTION_ID)"
+  managementGroupId: "$(ARM_MGMTGROUP_ID)"
+  modulesRepository: "$(modulesRepository)"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:59]:
```suggestion
  modulePath: '$(modulePath)'
  parametersRepository: '$(Build.Repository.Name)'
  location: '$(location)'
  resourceGroupName: '$(resourceGroupName)'
  subscriptionId: '$(ARM_SUBSCRIPTION_ID)'
  managementGroupId: '$(ARM_MGMTGROUP_ID)'
  modulesRepository: '$(modulesRepository)'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 5]
```diff
@@ -44,22 +44,22 @@
...
   # Azure PowerShell Version parameter
-  azurePowerShellVersion: '$(azurePowerShellVersion)'
-  preferredAzurePowerShellVersion: '$(preferredAzurePowerShellVersion)'
+  azurePowerShellVersion: "$(azurePowerShellVersion)"
+  preferredAzurePowerShellVersion: "$(preferredAzurePowerShellVersion)"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:62]:
```suggestion
  azurePowerShellVersion: '$(azurePowerShellVersion)'
  preferredAzurePowerShellVersion: '$(preferredAzurePowerShellVersion)'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 6]
```diff
@@ -81,7 +81,7 @@ jobs:
           - ${{ each checkoutRepository in parameters.checkoutRepositories }}:
               - checkout: ${{ checkoutRepository }}
                 fetchDepth: 1 # the depth of commits to ask Git to fetch; if not set defaults to no limit
-                path: 's/${{ checkoutRepository }}'
+                path: "s/${{ checkoutRepository }}"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:84]:
```suggestion
                path: 's/${{ checkoutRepository }}'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 7]
```diff
@@ -107,7 +107,7 @@ jobs:
       # [Agent] Prepare environment
       #----------------------------
       - task: PowerShell@2
-        displayName: 'Setup agent'
+        displayName: "Setup agent"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:110]:
```suggestion
        displayName: 'Setup agent'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 8]
```diff
@@ -128,7 +128,7 @@ jobs:
       # [Module Pester Test] task(s)
       #-----------------------------
       - task: AzurePowerShell@5
-        displayName: 'Run general tests via (Pester) via connection [${{ parameters.serviceConnection }}]'
+        displayName: "Run general tests via (Pester) via connection [${{ parameters.serviceConnection }}]"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:131]:
```suggestion
        displayName: 'Run general tests via (Pester) via connection [${{ parameters.serviceConnection }}]'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 9]
```diff
@@ -187,11 +187,11 @@ jobs:
       - task: PublishTestResults@2
         displayName: Publish Test Results
         inputs:
-          testRunTitle: 'Global Module Tests'
+          testRunTitle: "Global Module Tests"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:190]:
```suggestion
          testRunTitle: 'Global Module Tests'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 10]
```diff
@@ -187,11 +187,11 @@ jobs:
...
           testResultsFormat: NUnit
           testResultsFiles: global-testResults.xml
           failTaskOnFailedTests: true
-          searchFolder: 'arm/.global'
+          searchFolder: "arm/.global"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:194]:
```suggestion
          searchFolder: 'arm/.global'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 11]
```diff
@@ -211,7 +211,7 @@ jobs:
           - ${{ each checkoutRepository in parameters.checkoutRepositories }}:
               - checkout: ${{ checkoutRepository }}
                 fetchDepth: 1 # the depth of commits to ask Git to fetch; if not set defaults to no limit
-                path: 's/${{ checkoutRepository }}'
+                path: "s/${{ checkoutRepository }}"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:214]:
```suggestion
                path: 's/${{ checkoutRepository }}'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 12]
```diff
@@ -237,7 +237,7 @@ jobs:
       # [Agent] Prepare environment
       #----------------------------
       - task: PowerShell@2
-        displayName: 'Setup agent'
+        displayName: "Setup agent"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:240]:
```suggestion
        displayName: 'Setup agent'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 13]
```diff
@@ -251,7 +251,7 @@ jobs:
       # [Module Pester Test] task(s)
       #-----------------------------
       - task: AzurePowerShell@5
-        displayName: 'Run API tests via (Pester) via connection [${{ parameters.serviceConnection }}]'
+        displayName: "Run API tests via (Pester) via connection [${{ parameters.serviceConnection }}]"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:254]:
```suggestion
        displayName: 'Run API tests via (Pester) via connection [${{ parameters.serviceConnection }}]'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 14]
```diff
@@ -295,10 +295,10 @@ jobs:
       - task: PublishTestResults@2
         displayName: Publish Test Results
         inputs:
-          testRunTitle: 'Global Module API Tests'
+          testRunTitle: "Global Module API Tests"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:298]:
```suggestion
          testRunTitle: 'Global Module API Tests'
```

File: .azuredevops/pipelineTemplates/jobs.validateModulePester.yml [hunk 15]
```diff
@@ -295,10 +295,10 @@ jobs:
...
           testResultsFormat: NUnit
           testResultsFiles: api-testResults.xml
           failTaskOnFailedTests: true
-          searchFolder: 'arm/.global'
+          searchFolder: "arm/.global"
```
Comment (MariusStorhaug) [inline .azuredevops/pipelineTemplates/jobs.validateModulePester.yml:302]:
```suggestion
          searchFolder: 'arm/.global'
```

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]
Added the suggestions

**Review by MrMCake — APPROVED** [review]

**Pull request merged**